src/transformers/tokenization_utils.py

```diff
@@ -507,18 +612,39 @@ def vocab_size(self) -> int:
     def is_fast(self):
         return False
 
+    @property
+    def max_len(self):
+        """ Kept here for backward compatibility.
+            Now renamed to `model_max_length` to avoid ambiguity.
+        """
+        return self.model_max_length
+
+    @property
+    def max_len_single_sentence(self):
+        return self.model_max_length - self.num_special_tokens_to_add(pair=False)
+
+    @property
+    def max_len_sentences_pair(self):
+        return self.model_max_length - self.num_special_tokens_to_add(pair=True)
+
     def get_vocab(self):
         """ Returns the vocabulary as a dict of {token: index} pairs. `tokenizer.get_vocab()[token]` is equivalent to `tokenizer.convert_tokens_to_ids(token)` when `token` is in the vocab. """
         raise NotImplementedError()
 
-    def __init__(self, max_len=None, **kwargs):
+    def __init__(self, model_max_length=None, **kwargs):
 
         super().__init__(**kwargs)
 
-        self.max_len = max_len if max_len is not None else int(1e12)
+        # For backward compatibility we fallback to set model_max_length from max_len if provided
+        model_max_length = model_max_length if model_max_length is not None else kwargs.pop("max_len", None)
+        self.model_max_length = model_max_length if model_max_length is not None else VERY_LARGE_INTEGER
```

Comment:
`max_len` is not `model_max_length` (with a read-only `max_len` property for backward compatibility)